teps/0008-api-spec.md

```diff
@@ -0,0 +1,115 @@
+
+By explicitly stating which portions of the API _must_ be supported to be considered a "conformant" Tekton implementation, which are optional nice-to-haves, and which are explicitly not recommended or expected to be supported, the API is easier to understand and adopt for users, operators, platform providers, and the community as a whole.
+
+### Goals
+
+Produce and maintain a document in the [tektoncd/pipeline](https://github.com/tektoncd/pipeline) repo documenting the v1beta1 Tekton Pipelines API, including which fields are required, recommended and not recommended to be supported by a conformant Tekton Pipelines implementation.
+
+This will serve as documentation to end users and client authors (including Tekton's own CLI and Dashboard) about which features it should expect to be supported.
+
+Document the lifecycle of a Tekton Pipelines API feature, through alpha to beta to GA, and possibly separately to being required to satisfy API spec conformance. This process should cover how to communicate to clients and existing implementations that a new field will newly be considered required.
+
+Produce and maintain a conformance test suite which can be run against a candidate Tekton implementation to determine whether it satisfies the documented spec. This test suite should be regularly run against a platform to guard against regressions.
+
+### Non-Goals
+
+Document the [runtime contract](https://tekton.dev/docs/pipelines/container-contract/), or prescribe specific implementations of supporting services such as access control, observability and resource management.
+
+Document the [Tekton Triggers API](https://github.com/tektoncd/pipeline). This API surface should be similarly specified some time after it graduates to v1beta1 or v1.
+
+Document requirements or recommendations for the Tekton CLI or Dashboard projects. Requirements or recommendations may be documented in the future with those teams' input.
+
+## Requirements
+
+Acknowledgement and approval by Tekton's governance board, and stakeholders in the CLI and Dashboard projects that an API spec will be forthcoming. Discussion about specifics of the API spec can happen in the PR that proposes it.
+
+## Proposal
+
+The first iteration of the API spec should require the bare minimum functionality from a Tekton implementation -- that is, `TaskRun`s can be specified with `TaskSpec`s defined inline to specify `steps`. In addition to `steps`, a minimal `TaskRun` should support `params` and `results`, `workspaces`, and associated `status` fields. (The full list will be proposed in the initial draft API spec).
+
+Task references, including `ClusterTask`s and OCI image bundles, will not be included in the initial version, nor will `Pipeline`s or `PipelineRun`s, `Condition`s, etc. -- these will be added as appropriate in later iterations.
```

Comment:
Omitting references is a pretty large gap - it essentially means most of the catalog can't be used. Should we provide a catalog of inline taskruns?

Reply:
This is a temporary limitation, to reduce the scope of the initial spec surface. We absolutely fully intend to suppot the catalog in the fullness of time.

I think there's a discussion we could have about whether platforms _must_ support it with in-service Task/Pipeline CRUD or whether they _may_ support it only via OCI bundles (#137), and tbh I hope having that discussion outside the initial spec proposal (i.e., "should we _have_ a spec?") makes this much easier to accept.